frontend/src/modules/InplaceVolumetricsPlot/settings/settings.tsx

```diff
@@ -0,0 +1,177 @@
+
+import { useAtom, useAtomValue, useSetAtom } from "jotai";
+
+import {
+    userSelectedColorByAtom,
+    userSelectedEnsembleIdentsAtom,
+    userSelectedFluidZonesAtom,
+    userSelectedIdentifiersValuesAtom,
+    userSelectedPlotTypeAtom,
+    userSelectedResultName2Atom,
+    userSelectedResultNameAtom,
+    userSelectedSelectorColumnAtom,
+    userSelectedSubplotByAtom,
+    userSelectedTableNamesAtom,
+} from "./atoms/baseAtoms";
+import {
+    selectedColorByAtom,
+    selectedEnsembleIdentsAtom,
+    selectedFluidZonesAtom,
+    selectedIdentifiersValuesAtom,
+    selectedResultName2Atom,
+    selectedResultNameAtom,
+    selectedSelectorColumnAtom,
+    selectedSubplotByAtom,
+    selectedTableNamesAtom,
+    tableDefinitionsAccessorAtom,
+} from "./atoms/derivedAtoms";
+import { tableDefinitionsQueryAtom } from "./atoms/queryAtoms";
+import { makeColorByOptions, makeSubplotByOptions } from "./utils/plotDimensionUtils";
+
+import { Interfaces } from "../interfaces";
+import { PlotType, plotTypeToStringMapping } from "../typesAndEnums";
+
+export function Settings(props: ModuleSettingsProps<Interfaces>): React.ReactNode {
+    const ensembleSet = useEnsembleSet(props.workbenchSession);
+    const tableDefinitionsQueryResult = useAtomValue(tableDefinitionsQueryAtom);
+    const tableDefinitionsAccessor = useAtomValue(tableDefinitionsAccessorAtom);
+
+    const selectedEnsembleIdents = useAtomValue(selectedEnsembleIdentsAtom);
+    const setSelectedEnsembleIdents = useSetAtom(userSelectedEnsembleIdentsAtom);
+
+    const selectedTableNames = useAtomValue(selectedTableNamesAtom);
+    const setSelectedTableNames = useSetAtom(userSelectedTableNamesAtom);
+
+    const selectedSelectorColumn = useAtomValue(selectedSelectorColumnAtom);
+    const setSelectedSelectorColumn = useSetAtom(userSelectedSelectorColumnAtom);
+
+    const selectedFluidZones = useAtomValue(selectedFluidZonesAtom);
+    const setSelectedFluidZones = useSetAtom(userSelectedFluidZonesAtom);
+
+    const selectedIdentifiersValues = useAtomValue(selectedIdentifiersValuesAtom);
+    const setSelectedIdentifiersValues = useSetAtom(userSelectedIdentifiersValuesAtom);
+
+    const selectedResultName = useAtomValue(selectedResultNameAtom);
+    const setSelectedResultName = useSetAtom(userSelectedResultNameAtom);
+
+    const selectedResultName2 = useAtomValue(selectedResultName2Atom);
+    const setSelectedResultName2 = useSetAtom(userSelectedResultName2Atom);
+
+    const selectedSubplotBy = useAtomValue(selectedSubplotByAtom);
+    const setSelectedSubplotBy = useSetAtom(userSelectedSubplotByAtom);
+
+    const selectedColorBy = useAtomValue(selectedColorByAtom);
+    const setSelectedColorBy = useSetAtom(userSelectedColorByAtom);
+
+    const [selectedPlotType, setSelectedPlotType] = useAtom(userSelectedPlotTypeAtom);
+
+    function handleFilterChange(newFilter: InplaceVolumetricsFilter) {
```

Comment:
This statement is not true per se. `useCallback` should be used whenever it makes a difference or is required in order to prevent costly unnecessary re-renders (especially when it's a dependency somewhere down the line - e.g. memo/useEffect/useCallback). In order to analyze if a wrapping is required, it makes sense to look at how the callback is consumed, how often it gets renewed (i.e. its defining component gets re-rendered), and what dependencies itself has. Wrapping everything in `useCallback` makes code worse readable and comes at the cost of memory-usage. Same accounts for `useMemo`. Very useful hooks but IMO not to be just thrown at your code (I know some teams do).

At the bottom of this page is a deep dive: https://react.dev/reference/react/useCallback

Reply:
Yeah, turns out I misunderstood the React lifecycle slightly; I thought child components **only** re-rendered if **their own** props changed (I think this is the case in Vue), but that wasn't true. So feel free to ignore this feedback